### PR TITLE
Remove raft-ann-bench

### PR DESCRIPTION
**Pull request by cjnolet**
This PR removes raft-ann-bench from the conda packages, build system, and documentation. This removal was previously announced for the 24.12 release in #2448.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -68,23 +63,13 @@ option(DISABLE_OPENMP "Disable OpenMP" OFF)
 option(RAFT_NVTX "Enable nvtx markers" OFF)
 
 set(RAFT_COMPILE_LIBRARY_DEFAULT OFF)
-if((BUILD_TESTS
-    OR BUILD_PRIMS_BENCH
-    OR BUILD_ANN_BENCH
-   )
-   AND NOT BUILD_CPU_ONLY
-)
+if((BUILD_TESTS OR BUILD_PRIMS_BENCH))
```
Comment (bdice) [inline cpp/CMakeLists.txt:66]:
```suggestion
if(BUILD_TESTS OR BUILD_PRIMS_BENCH)
```

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -98,19 +83,14 @@ message(VERBOSE "RAFT: Building optional components: ${raft_FIND_COMPONENTS}")
 message(VERBOSE "RAFT: Build RAFT unit-tests: ${BUILD_TESTS}")
 message(VERBOSE "RAFT: Building raft C++ benchmarks: ${BUILD_PRIMS_BENCH}")
 message(VERBOSE "RAFT: Building ANN benchmarks: ${BUILD_ANN_BENCH}")
```
Comment (bdice) [inline cpp/CMakeLists.txt:85]:
```suggestion
```

File: build.sh
Comment (bdice) [inline build.sh:1]: More edits: remove line 72 `BUILD_CPU_ONLY=OFF` and line 406 `-DBUILD_CPU_ONLY=${BUILD_CPU_ONLY} \`

Same for `BUILD_ANN_BENCH` on lines 71 and 405.

**Review by bdice — COMMENTED** [review]
There are a few more instances of `ANN_BENCH` in `docs/source/` and `.github/workflows/pr.yaml` and a few other spots.

**Comment by bdice** [comment]
@cjnolet I think this PR should be ready once we get CI passing. It turns out there were quite a few pieces we could prune as a result of removing raft-ann-bench. I ran devcontainer builds locally and it seems fine. I see the pip devcontainer failed, I will rerun it soon. I don't think the failure is due to these changes.

**Review by benfred — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Pull request merged**